wire/msgaddr2_test.go

```diff
@@ -0,0 +1,664 @@
+package wire
+
+import (
+	"bytes"
+	"encoding/hex"
+	"io"
+	"net"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+// hexToBytes converts the passed hex string into bytes and will panic if there
+// is an error.  This is only provided for the hard-coded constants so errors in
+// the source code can be detected. It will only (and must only) be called with
+// hard-coded values.
+func hexToBytes(s string) []byte {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		panic("invalid hex in source file: " + s)
+	}
+	return b
+}
+
+// TestAddrV2 tests the MsgAddrV2 API.
+func TestAddrV2(t *testing.T) {
+	pver := ProtocolVersion
+
+	// Ensure the command is expected value.
+	wantCmd := "addrv2"
+	msg := NewMsgAddrV2()
+	if cmd := msg.Command(); cmd != wantCmd {
+		t.Errorf("NewMsgAddrV2: wrong command - got %v want %v",
+			cmd, wantCmd)
+	}
+
+	// Ensure max payload is expected value for latest protocol version.
+	// Num addresses (varInt) + num * (time + services + networkID + addr len + addr + port)
+	wantPayload := uint32(9 + 1000*(4+9+1+9+512+2))
```

Comment:
Since the `addr` is at most 512 bytes, the `addr len` is at most 3 bytes in `CompactSize`.
(A 1 byte length selection prefix, plus a 2 byte integer.)

But it's ok if you don't want to account for that here.

```suggestion
	wantPayload := uint32(9 + 1000*(4+9+1+3+512+2))
```

Reply:
Done in 2c72d105